chardet/charsetprober.py

```diff
@@ -79,8 +78,8 @@ def filter_international_words(buf):
 
         This filter applies to all scripts which do not use English characters.
         """
-        filtered = BytesIO()
 
+        out = b''
```

Comment:
Umm... I just ran this and it completely confirmed my suspicions.

The `BytesIO` part (with Python 3) finished in 3.5 seconds, and the part using `bytes` with concatenation has been running for well over a minute and is using 2.5GB of RAM.